pkg/monitor/datapath_trace.go

```diff
@@ -17,13 +17,15 @@ package monitor
 import (
 	"encoding/json"
 	"fmt"
+	"net"
 
+	"github.com/cilium/cilium/common/types"
 	"github.com/cilium/cilium/pkg/monitor/api"
 )
 
 const (
 	// TraceNotifyLen is the amount of packet data provided in a trace notification
-	TraceNotifyLen = 32
+	TraceNotifyLen = 48
```

Comment:
Do we enforce recreation of the perf event channel on startup based upon this? Otherwise is there a guarantee upon upgrade that the new agent-side monitor won't get confused by the existing datapath pumping old-style trace notifications into the perf events channel?

Reply:
Good point :-/

Reply:
It must be. The memory is owned by the user space process so it must get re-created when the agent restarts, no?

Reply:
I've tested this. I did not observe any problems restarting between master and this PR branch.